src/controller/proxy/controller.go

```diff
@@ -161,26 +164,39 @@ func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo,
 	}
 
 	var content []byte
-	if c.cache != nil {
-		err = c.cache.Fetch(getManifestListKey(art.Repository, string(desc.Digest)), &content)
-		if err == nil {
-			log.Debugf("Get the manifest list with key=cache:%v", getManifestListKey(art.Repository, string(desc.Digest)))
-			return true, &ManifestList{content, string(desc.Digest), manifestlist.MediaTypeManifestList}, nil
-		}
+	var contentType string
+	if c.cache == nil {
+		return a != nil && string(desc.Digest) == a.Digest, nil, nil // digest matches
+	}
+
+	err = c.cache.Fetch(manifestListKey(art.Repository, string(desc.Digest)), &content)
+	if err != nil {
 		if err == cache.ErrNotFound {
-			log.Debugf("Digest is not found in manifest list cache, key=cache:%v", getManifestListKey(art.Repository, string(desc.Digest)))
+			log.Debugf("Digest is not found in manifest list cache, key=cache:%v", manifestListKey(art.Repository, string(desc.Digest)))
 		} else {
 			log.Errorf("Failed to get manifest list from cache, error: %v", err)
 		}
+		return a != nil && string(desc.Digest) == a.Digest, nil, nil
 	}
-	return a != nil && string(desc.Digest) == a.Digest, nil, nil // digest matches
+	err = c.cache.Fetch(manifestListContentTypeKey(art.Repository, string(desc.Digest)), &contentType)
+	if err != nil {
+		log.Debugf("failed to get the manifest list content type, not use local.")
```

Comment:
log err here?

Reply:
and why not reture err?

Reply:
return error will cause the error to return a non 200 http code to the client-side and pull fail, current situation, the error to access cache can be logged and ignored.